### DISCUSSION TITLE
Can stdlib be used to unzip npz format?

### DISCUSSION THREAD

**Discussion by jalvesz** [fortran-lang/stdlib#751] [General]
Hi,
I saw that stdlib has a facility function to open npy files, I was wondering if this could be extended to treat npz? https://numpy.org/doc/stable/reference/generated/numpy.savez.html

**Comment by perazz** [comment]
@minhqdao

**Comment by minhqdao** [comment]
I'll have a look into it

**Comment by jalvesz** [comment]
Hi, I wanted to share a thought and ask for opinions. I'm using the npy reader and for my application I can run the process on multiple processes with MPI. I need all processes to retrieve the same data from my list of .npy at the very beginning. So for testing I added the `shared` option to the `open` statement as I'm using intel compilers, it works perfect.

So, given that each compiler has different extensions for shared IO, I was wondering if it would be worth adding this features as an optional argument to `load_npy` which would then be translated to the appropriate option compiler-wise. Any thoughts?

Thanks

**Reply by minhqdao** [reply]
Yes, if it is feasible that `load_npy` sets the right options for different compilers (not sure) then I agree that this could be a useful feature.
